File: proj.android_studio/ee-x-facebook-ads/src/main/java/com/ee/facebook/FacebookNativeAdBuilder.java
package com.ee.facebook;

import android.support.annotation.NonNull;

/**
 * Created by Zinge on 10/9/17.
 */

class FacebookNativeAdBuilder {
    String adId;
    String layoutName;
    String icon;
    String title;
    String media;
    String socialContext;
    String adChoices;
    String body;
    String action;

    FacebookNativeAdBuilder setAdId(@NonNull String value) {
        adId = value;
        return this;
    }

    FacebookNativeAdBuilder setLayoutName(@NonNull String value) {
        layoutName = value;
        return this;
    }

    FacebookNativeAdBuilder setIcon(@NonNull String key) {
        icon = key;
        return this;
    }

    FacebookNativeAdBuilder setTitle(@NonNull String key) {
        title = key;
        return this;
    }

    FacebookNativeAdBuilder setMedia(@NonNull String key) {
        media = key;
        return this;
    }

    FacebookNativeAdBuilder setSocialContext(@NonNull String key) {
        socialContext = key;
        return this;
    }

    FacebookNativeAdBuilder setAdChoices(@NonNull String key) {
        adChoices = key;
        return this;
    }

    FacebookNativeAdBuilder setBody(@NonNull String key) {
        body = key;
        return this;
    }

    FacebookNativeAdBuilder setAction(@NonNull String key) {
        action = key;
        return this;
    }
}

File: src/ee/facebook/FacebookNativeAdBuilder.cpp
//
//  FacebookNativeAdBuilder.cpp
//  ee_x
//
//  Created by Zinge on 10/9/17.
//
//

#include "ee/facebook/FacebookNativeAdBuilder.hpp"

namespace ee {
namespace facebook {
using Self = FacebookNativeAdBuilder;

Self::FacebookNativeAdBuilder() {}

Self::~FacebookNativeAdBuilder() {}

Self& Self::setAdId(const std::string& adId) {
    adId_ = adId;
    return *this;
}

Self& Self::setLayoutName(const std::string& layoutName) {
    layoutName_ = layoutName;
    return *this;
}

Self& Self::setIcon(const std::string& key) {
    icon_ = key;
    return *this;
}

Self& Self::setTitle(const std::string& key) {
    title_ = key;
    return *this;
}

Self& Self::setMedia(const std::string& key) {
    media_ = key;
    return *this;
}

Self& Self::setSocialContext(const std::string& key) {
    socialContext_ = key;
    return *this;
}

Self& Self::setAdChoices(const std::string& key) {
    adChoices_ = key;
    return *this;
}

Self& Self::setBody(const std::string& key) {
    body_ = key;
    return *this;
}

Self& Self::setAction(const std::string& key) {
    action_ = key;
    return *this;
}
} // namespace facebook
} // namespace ee
